voting/orbs-pos-data/src/__tests__/staking-service-mock.test.ts

```diff
@@ -24,58 +26,137 @@ function testTxCreatingMethods() {
   });
 }
 
-function testDataReadingMethods() {
-  describe(`Data reading methods`, () => {
-    let stakingService: StakingServiceMock;
+function testDataReadingMethodsWitchRealAPI() {
+  describe('Data reading methods with real writing API', () => {
+    const ownerAddress = '0xowner';
+    let stakingService: IStakingService;
 
     beforeEach(() => {
-      stakingService = new StakingServiceMock(false);
+      stakingService = new StakingServiceMock(true);
+      stakingService.setFromAccount(ownerAddress);
+    });
+
+    it(`Should allow to set and get stake balance`, async () => {
+      const beforeResult = await stakingService.readStakeBalanceOf(ownerAddress);
+      expect(beforeResult).toEqual('0');
+
+      await stakingService.stake(500000);
+
+      const afterResult = await stakingService.readStakeBalanceOf(ownerAddress);
+      expect(afterResult).toEqual('500000');
+    });
+
+    it(`Should return the correct cooldown status after stake, unstake and restake`, async () => {
+      const beforeResult = await stakingService.readTotalStakedTokens();
+      expect(beforeResult, 'Initial amount should be 0').toEqual('0');
+
+      // Ensure that one stake updates
+      await stakingService.stake(111111);
+
+      const afterResult = await stakingService.readTotalStakedTokens();
+      expect(afterResult, 'Should be equal to staked amount').toEqual('111111');
+
+      // Ensure several stakes
+      await stakingService.stake(222222);
+      await stakingService.stake(333333);
+
+      const afterSeveralStakesResult = await stakingService.readTotalStakedTokens();
+      expect(afterSeveralStakesResult, 'Should be equal to the sum of all previous "stake" orders').toEqual('666666');
+
+      // Unstake
+      await stakingService.unstake(222222);
+
+      const resultAfterUnstaking = await stakingService.readTotalStakedTokens();
+      expect(resultAfterUnstaking, 'Should subtract the unstaked tokens from the total amount').toEqual('444444');
+
+      // Restake
+      await stakingService.restake();
+      const resultAfterRestake = await stakingService.readTotalStakedTokens();
+      expect(resultAfterRestake, 'Should add all restaked tokens to the total count').toEqual('666666');
+    });
+
+    it(`Should return the correct cooldown status after stake, unstake and restake`, async () => {
+      const cooldownPeriodTimeInSeconds = 100;
+      (stakingService as StakingServiceMock).setCooldownTime(cooldownPeriodTimeInSeconds);
```

Comment:
Start the mock with now+day, and the user of the testkit can override it with setNextCooldownTime in your tests

Reply:
Done